iceoryx_hoofs/concurrent/buffer/include/iox/detail/spsc_fifo.inl

```diff
@@ -27,33 +27,47 @@ namespace concurrent
 template <class ValueType, uint64_t Capacity>
 inline bool SpscFifo<ValueType, Capacity>::push(const ValueType& value) noexcept
 {
-    if (is_full())
+    // Memory order relaxed is enough since:
+    // - there is no concurrent access to this method
+    // - the load statement cannot be reordered with writing m_data otherwise it would not compile
+    auto currentWritePos = m_writePos.load(std::memory_order_relaxed);
+    bool isFull = (currentWritePos ==
+                   // Memory order acquire is needed to ensure happens-before relationship
```

Comment:
I think it would be good to also add the corresponding sync point. In this case it would be the `store  with release ordering on m_readPos in the pop method`.

Similar with other non-relaxed memory orderings. It is essential to know the counterparts of the sync operation.